### ISSUE TITLE
[BUG] Timestamps are Date while local object is mutated

### ISSUE THREAD

**Issue by henhal** [dynamoose/dynamoose#1163]
### Summary:
Timestamps are in epochs (numbers), but when calling `save()`, the mutated object gets a `Date` object instead. The resulting object returned in the promise has number timestamps again.

#### Schema
```
N/A
```

#### Model
```
const m = dynamoose.model("Foo", new dynamoose.Model{id: String}, {timestamps: true});
const foo1 = await m.create({id: 'abc'});
// now,foo1.updatedAt is a number
console.log('foo1 after create:', foo1);

foo1.hello = 'world'; 
const foo2 = await foo1.save();
// now, foo1.updatedAt is a Date object, but foo2.updatedAt is a number
console.log('foo1 after mutation:', foo1);
console.log('foo2 after mutation:', foo2);
```

#### General
```
N/A
```

### Current output and behavior (including stack trace):

```
foo1 after create: {
  "id": "abc",
  "createdAt": 1616084000594,
  "updatedAt": "1616084000594",
}
foo1 after mutation: {
  "id": "abc",
  "createdAt": 1616084000594,
  "updatedAt": "2021-03-18T16:13:20.646Z",
}
foo2 after mutation: {
  "id": "abc",
  "createdAt": 1616084000594,
  "updatedAt": 1616084000646,
}
```

### Expected output and behavior:

Timestamps should be consistently typed even for mutated local objects that aren't saved yet.

### Environment:

Operating System: Ubuntu
Operating System Version: 18.04
Node.js version (`node -v`): 14.2.0
NPM version: (`npm -v`): 6.14.4
Dynamoose version: 2.7.3

### Other:
- [X] I have read through the Dynamoose documentation before posting this issue
- [X] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [X] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [X] I have tested the code provided and am confident it doesn't work as intended
- [X] I have filled out all fields above
- [X] I am running the latest version of Dynamoose

**Comment by goatandsheep** [comment]
I believe this is because TimestampObject is type `String` not `Date` https://github.com/dynamoose/dynamoose/blob/40443acc6f7a07681592f294d3893902c3652e34/lib/Schema.ts#L247

**Comment by fishcharlie** [comment]
> I believe this is because TimestampObject is type `String` not `Date`
> 
> https://github.com/dynamoose/dynamoose/blob/40443acc6f7a07681592f294d3893902c3652e34/lib/Schema.ts#L247

@goatandsheep That is the TypeScript interface for Schema settings. Doesn't have anything to do with the documents.

I do think this will be resolved in #1258 tho.
